lib/internal/validators.js

```diff
@@ -218,41 +218,47 @@ function validateBoolean(value, name) {
     throw new ERR_INVALID_ARG_TYPE(name, 'boolean', value);
 }
 
-/**
- * @param {any} options
- * @param {string} key
- * @param {boolean} defaultValue
- * @returns {boolean}
- */
-function getOwnPropertyValueOrDefault(options, key, defaultValue) {
-  return options == null || !ObjectPrototypeHasOwnProperty(options, key) ?
-    defaultValue :
-    options[key];
-}
+const kValidateObjectAllowNullable = 1 << 0;
+const kValidateObjectAllowArray = 1 << 1;
+const kValidateObjectAllowFunction = 1 << 2;
 
 /**
  * @callback validateObject
  * @param {*} value
  * @param {string} name
- * @param {{
- *   allowArray?: boolean,
- *   allowFunction?: boolean,
- *   nullable?: boolean
- * }} [options]
+ * @param {number} [options]
  */
 
 /** @type {validateObject} */
 const validateObject = hideStackFrames(
-  (value, name, options = null) => {
-    const allowArray = getOwnPropertyValueOrDefault(options, 'allowArray', false);
-    const allowFunction = getOwnPropertyValueOrDefault(options, 'allowFunction', false);
-    const nullable = getOwnPropertyValueOrDefault(options, 'nullable', false);
-    if ((!nullable && value === null) ||
-        (!allowArray && ArrayIsArray(value)) ||
-        (typeof value !== 'object' && (
-          !allowFunction || typeof value !== 'function'
-        ))) {
-      throw new ERR_INVALID_ARG_TYPE(name, 'Object', value);
+  (value, name, options = 0) => {
```

Comment:
Please name the 0 here as a constant e.g. `options = kNone` and `const kNone = 0;` above to make it clear the 0 is a bit field here